packages/@aws-cdk/core/lib/stack.ts

```diff
@@ -436,6 +443,7 @@ export class Stack extends Construct implements ITaggable {
     this._versionReportingEnabled = (props.analyticsReporting ?? this.node.tryGetContext(cxapi.ANALYTICS_REPORTING_ENABLED_CONTEXT))
       && !this.nestedStackParent;
 
+    // TODO: want to add my new synthesizer here but can't, because that's a dependency cycle, right?
```

Comment:
@rix0rrr this feels like I can't actually create the new synthesizer in a separate module that depends on @aws-cdk/core, am i wrong?